engine/src/main/java/org/terasology/engine/TerasologyEngine.java

```diff
@@ -131,11 +128,14 @@
     private GameState pendingState;
     private Set<StateChangeSubscriber> stateChangeSubscribers = Sets.newLinkedHashSet();
 
-    private enum EngineState {UNINITIALIZED, INITIALIZED, RUNNING, DISPOSED}
+    private EngineStatus status = StandardGameStatus.UNSTARTED;
+    private final List<EngineStatusSubscriber> statusSubscriberList = new CopyOnWriteArrayList<>();
 
-    private EngineState engineState = EngineState.UNINITIALIZED;
-
-    private final TaskMaster<Task> commonThreadPool = TaskMaster.createFIFOTaskMaster("common", MAX_NUMBER_THREADS);
+    private transient boolean initialised;
+    private transient boolean shutdownRequested;
+    private transient boolean shuttingDown;
+    private transient boolean running;
+    private transient boolean disposed;
```

Comment:
You are right that it can be simplified, but I think I'll go further.

The only state is running - true or false. No initialised, no disposed. Part of shutdown will be disposal. Once run the engine can be rerun, or discarded.

Request shutdown will remain - it isn't a state, just a flag to trigger shutdown. Initialised/shutting down will be removed.